rslib/src/scheduler/queue/builder/mod.rs

```diff
@@ -482,4 +484,37 @@ mod test {
         // include the buried card.
         assert_eq!(col.card_queue_len(), old_queue_len - 1);
     }
+
+    #[test]
+    fn new_cards_may_ignore_review_limit() {
+        let mut col = Collection::new_v3();
+        col.update_default_deck_config(|config| {
+            config.new_ignore_review_limit = true;
+            config.reviews_per_day = 0;
+        });
+        NoteAdder::basic(&mut col).add(&mut col);
+
+        // review limit doesn't apply to new card
+        assert_eq!(col.card_queue_len(), 1);
+    }
+
+    #[test]
+    fn review_limit_only_applies_to_new_cards_from_the_same_deck() {
+        let mut col = Collection::new_v3();
+        // parent deck with more new than reviews, but new cards ignore the review limit
+        let parent = DeckAdder::new("parent")
+            .with_config(|config| {
+                config.inner.reviews_per_day = 0;
+                config.inner.new_per_day = 1;
+                config.inner.new_ignore_review_limit = true;
+            })
```

Comment:
These sort of helpers are great, and should help encourage us to write more tests.